pinot-segment-local/src/main/java/org/apache/pinot/segment/local/data/manager/TableDataManager.java

```diff
@@ -201,4 +201,18 @@ void addOrReplaceSegment(String segmentName, IndexLoadingConfig indexLoadingConf
    * @return List of {@link SegmentErrorInfo}
    */
   Map<String, SegmentErrorInfo> getSegmentErrors();
+
+  /**
+   * Interface to place actions to be performed when a Segment goes from CONSUMING to ONLINE.
+   *
+   * @param segmentNameStr name of segment for which the state change is being handled
+   */
+  default void onConsumingToOnline(String segmentNameStr) { };
+
+  /**
+   * Interface to place actions to be performed when a Segment goes from CONSUMING to DROPPED.
+   *
+   * @param segmentNameStr name of segment for which the state change is being handled
+   */
+  default void onConsumingToDropped(String segmentNameStr) { };
 }
```

Comment:
Can we use a single method for handling state transition notifications?
```
default void onSegmentStateTransition(String segmentName, String fromState, String toState) { }
```
It will be easier to enhance it rather than add more interface methods?

Reply:
Seems like a good suggestion to keep the high level interface simpler, working on this.

Reply:
We tried this but it made things a little more complicated so the change was rolled back